frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1130,6 +1134,41 @@ func isReachable(from, to *dispatchState) (ret bool) {
 	return false
 }
 
+func hasCircularDependency(states []*dispatchState) (bool, *dispatchState) {
```

Comment:
Can we have UT?

Reply:
sure, added "TestDockerfileCircularDependencies"